onmt/model_builder.py

```diff
@@ -150,16 +150,21 @@ def load_test_model(opt, model_path=None):
     # Avoid functionality on inference
     model_opt.update_vocab = False
 
-    model = build_base_model(model_opt, vocabs, use_gpu(opt), checkpoint,
-                             opt.gpu)
+    model = build_base_model(model_opt, vocabs, checkpoint)
+
     if opt.fp32:
         model.float()
+    elif opt.fp16:
+        model.half()
     elif opt.int8:
         if opt.gpu >= 0:
             raise ValueError(
                 "Dynamic 8-bit quantization is not supported on GPU")
         torch.quantization.quantize_dynamic(model, inplace=True)
 
+    if use_gpu(opt) and opt.gpu >= 0:
```

Comment:
Can l155 to l166 be moved to build_base_model ?

Reply:
inference options only.